include/rocksdb/concurrent_task_limiter.h

```diff
@@ -0,0 +1,45 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+//
+// Copyright (c) 2011 The LevelDB Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file. See the AUTHORS file for names of contributors.
+
+#pragma once
+
+#include "rocksdb/env.h"
+#include "rocksdb/statistics.h"
+
+namespace rocksdb {
+
+class ConcurrentTaskLimiter {
+ public:
+
+  virtual ~ConcurrentTaskLimiter() {}
+
+  virtual const std::string& GetName() const = 0;
+
+  virtual void SetMaxOutstandingTask(int32_t limit) = 0;
+
+  virtual void ResetMaxOutstandingTask() = 0;
+  
+  virtual int32_t GetOutstandingTask() const = 0;
+
+  virtual bool GetToken(bool force, int32_t& tasks) = 0;
```

Comment:
We follow Google C++ Style, which disallows non-const reference in function arguments: https://google.github.io/styleguide/cppguide.html#Reference_Arguments

Here we need to pass a pointer.